language/move-prover/tests/sources/functional/bitwise_features.move

```diff
@@ -0,0 +1,52 @@
+// exclude_for: cvc5
+address 0x123 {
+
+module TestFeatures {
+    use std::vector;
+
+    /// The enabled features, represented by a bitset stored on chain.
+    struct Features has key {
+        features: vector<u8>,
+    }
+
+    spec Features {
+        pragma bv=b"0";
+    }
+
+    /// Helper to check whether a feature flag is enabled.
+    fun contains(features: &vector<u8>, feature: u64): bool {
+        let byte_index = feature / 8;
+        let bit_mask = 1 << ((feature % 8) as u8);
+        byte_index < vector::length(features) && ((*vector::borrow(features, byte_index))) & bit_mask != 0
+    }
+
+    spec contains {
+        pragma bv=b"0";
+        aborts_if false;
+        ensures result == ((feature / 8) < vector::length(features) && ((int2bv((((1 as u8) << ((feature % (8 as u64)) as u64)) as u8)) as u8)
+            & vector::borrow(features, feature / 8) as u8) > (0 as u8));
```

Comment:
`vector::borrow(features, feature / 8)` can be written as `features[feature/8]` in the spec, which is perhaps a bit more readable.